torch/csrc/Module.cpp

```diff
@@ -1544,6 +1545,12 @@ Call this whenever a new thread is created in order to propagate values from
       "_set_conj", [](const at::Tensor& x, bool conj) { x._set_conj(conj); });
   py_module.def(
       "_set_neg", [](const at::Tensor& x, bool neg) { x._set_neg(neg); });
+  py_module.def("_get_tensor_mathbits", &torch::jit::getTensorMathBits);
+  py_module.def(
+      "_set_tensor_mathbits",
+      static_cast<void (*)(
+          const at::Tensor&, std::unordered_map<int8_t, bool>)>(
```

Comment:
I'm not a big fan of this being a map from ints to bools. If someone is inspecting the Pickle data by hand, they will see an opaque numeric constant that they will have to grovel in the C++ enum definition to resolve the name of.  Also, you don't even say on the enum that these numbers are serialized and must be kept consistent.

Reply:
I think a more forward looking format would be to have a dictionary from string to bool. This can be conceptualized as arbitrary extra metadata attached to the tensor, and lets us add more properties to it in the future if needed. Additionally, you should NOT add entries to the dictionary if they are the default values.

Reply:
I did start with map of `<string, bool>` but `string` seemed like an overkill for `key`. But I think what you have mentioned makes sense. Have updated the type of map.

Thanks!